src/index/ivf_flat.rs

```diff
@@ -57,20 +57,18 @@ impl IVFFlatIndexImpl {
                 nlist as usize,
                 metric,
             ))?;
-            let own_fields_ = if own_fields { 1 } else { 0 };
-            faiss_IndexIVFFlat_set_own_fields(inner, own_fields_);
-
+            faiss_IndexIVFFlat_set_own_fields(inner, own_fields as i32);
```

Comment:
IIRC the effective argument type of `own_fields` is `c_int`, so we might as well follow that. Please try either `c_int::from(own_fields)` or `own_fields as c_int` if the former does not work.

Reply:
OK